### DISCUSSION TITLE
Why is device sync being deprecated?

### DISCUSSION THREAD

**Discussion by dlorddd** [realm/realm-dotnet#3676] [General]
Hi there,
I recently implemented realm and device sync in one of our products and find it to be working quite well. Seeing the deprecation mail a week after successful migration, I was a bit shocked. I started looking for alternatives and noted that there kind of are none. I am using Unity and synchronise data for multiple users in a local first fashion. Do you have - after deprecating this product - any advice on what to use instead? Should I opt out of MongoDB altogether?

**Comment by bdach** [comment]
The deprecation notice in README.md is in itself confusing, no doubt moreso due to the layers of branding involved.

> We announced the deprecation of Atlas Device Sync + Realm SDKs in September 2024.
> For a version of `realm-dotnet` without sync features, install version 20 or see the "community" branch.

From https://www.mongodb.com/docs/atlas/device-sdks/deprecation/:

> The Atlas Device SDKs are made up of two parts: the on-device database and Atlas Device Sync. The on-device database will continue to exist as an open source project.

So... if our project is using the offline parts of realm only and not sync, then is that what will "continue to exist"? Are there any maintainers lined up for the open source project or is it a giant unknown if realm will continue to be maintained going forward?

**Reply by papafe** [reply]
You are correct, it the "old Realm SDK" that will continue to be open source here on GitHub, but there is no plan to actively maintain it.

**Reply by bdach** [reply]
Disappointing news, but thanks for letting us know at least.

**Reply by peppy** [reply]
@papafe does this also apply to `realm-core` development? for instance, right now the latest realm version is unusable on android devices as it causes a [hard crash](https://github.com/realm/realm-core/issues/7950).

**Reply by charlesroddie** [reply]
A bit more info from another thread:

> https://github.com/realm/realm-dart/issues/1804#issuecomment-2426311892
> We're addressing production-impacting bugs reported via the [support portal](https://support.mongodb.com/)... Bugs reported on Github or ones that have workarounds will be evaluated on a case-by-case basis during our sync up meetings, but we expect not to work on the majority of those and instead rely on the community to resolve them.

Do you think that, for offline-only use, it's ok to update to 20.0.0 now? I assume that any community updates would continue the 20.0.0 line of releases? Don't want to get into a dead-end where I end up with unupdatable .realm files.

**Comment by papafe** [comment]
Hi @dlorddd,

You can take a look at [this page](https://www.mongodb.com/docs/atlas/app-services/sync/device-sync-deprecation/#alternative-solutions) about alternative to solution to Device Sync.

**Comment by peachypixels** [comment]
Hello!

Given the deprecation news, am I right in saying that...

1) Realm Core is still going to be actively maintained by MongoDB?
2) Realm Dot Net is now going to be maintained as open source?
3) That the 15.N version will be retired in September 2025? (once sync services are deprecated)
4) That the 20.N version is the long term version? (for those not using sync services)
5) That the community branch is in-effect the 20.N version? (with device sync removed)

The announcement and documentation were a little vague in places, so I'm just trying to get a handle on the intention going forward so I can plan accordingly with my current (in beta) project.

Of course, I'm aware anything is subject to change and entirely dependant on open source contributors. But if anyone from MongoDB could shed a little more light, it would be hugely appreciated.

On a side note...

I just wanted to say thanks to anyone who has contributed to this project (and will do so in the future). I replaced a custom data store solution (for the project in question) with Realm and it was one of the best decisions made. It's been easy to use, stable & very performant. Not only that, but the developers were so incredibly responsive & helpful to issues raised.

It's a shame to see MongoDB part ways with it, but equally hopeful that it will prosper in the open source community.

**Reply by nirinchev** [reply]
Unfortunately, the first statement is imprecise. To add some nuance:
1. Realm Core is stewarded by MongoDB, but we're not actively contributing new functionality. Through Sept. 2025, we're committed to addressing critical bugs - after that, it'll depend on what type of bug it is and whether it's affecting customers who've received extended support. The team would be willing to assist the community by doing PR reviews and releases, but we don't have resources to add new functionaltiy ourselves.

Essentially, the goal is to maintain the entire Realm product (Core database + language SDKs) as a community-owned open source project. We realize that this means that (at least initially) there'll be fewer features being added than when MongoDB was funding the development, but we believe the product is in a fairly good place in terms of functionality and reliability, so the current version should be sufficient for the majority of use cases. At the same time, we're very open for contributions and if there are skilled community members who would like to get maintainer permissions for the repos, we'd be willing to work with them and make that happen.

**Reply by peachypixels** [reply]
Thanks for the clarification @nirinchev 

I take it therefore that my assumptions (items 2-5) are 'fairly' accurate?

**Reply by nirinchev** [reply]
Yeah, those are correct.

**Reply by charlesroddie** [reply]
> At the same time, we're very open for contributions and if there are skilled community members who would like to get maintainer permissions for the repos, we'd be willing to work with them and make that happen.

@nirinchev I would like to get maintainer permissions for myself and @foggyfinder at SummaticLtd . How could that happen?

**Reply by charlesroddie** [reply]
CC @papafe and also @AlexeyBuryanov who is publishing a [fixed version](https://github.com/realm/realm-dotnet/issues/3711#issuecomment-4124705476) while this repo is unmaintained.
